state/tstate/recorder_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package tstate
+
+import (
+	"context"
+	"crypto/sha256"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/keys"
+)
+
+func FuzzRecorderPermissionValidator(f *testing.F) {
+	for i := 0; i < 10; i++ {
+		shaBytes := sha256.Sum256([]byte{byte(i), byte(i >> 8)})
+		bytes := shaBytes[:]
+		for bytes[len(bytes)-1] > 16 {
+			shaBytes := sha256.Sum256([]byte{bytes[len(bytes)-2], bytes[len(bytes)-1]})
+			bytes = append(bytes, shaBytes[:]...)
+		}
+		f.Add(bytes)
+	}
+	f.Fuzz(
+		RecorderPermissionValidatorFuzzer,
+	)
+}
+
+func createKeys(prefix byte) (keylist [][]byte) {
+	for i := 0; i < 512; i++ {
+		shaBytes := sha256.Sum256([]byte{prefix, byte(i), byte(i >> 8)})
+		randNewKey := make([]byte, 30, 32)
+		copy(randNewKey, shaBytes[:])
+		keylist = append(keylist, keys.EncodeChunks(randNewKey, 1))
+	}
+	return
+}
+
+func createKeysValues(keys [][]byte) (out map[string][]byte) {
+	out = map[string][]byte{}
+	for i, key := range keys {
+		shaBytes := sha256.Sum256(key)
+		out[string(keys[i])] = shaBytes[:]
+	}
+	return out
+}
+
+func nextByte(randBytes []byte) (byte, []byte, bool) {
+	if len(randBytes) == 0 {
+		return 0, randBytes, true
+	}
+	return randBytes[0], randBytes[1:], false
+}
+
+func nextUint16(randBytes []byte) (uint16, []byte, bool) {
+	if len(randBytes) < 2 {
+		return 0, randBytes, true
+	}
+	return uint16(randBytes[0]) | (uint16(randBytes[0]) << 8), randBytes[2:], false
+}
+
+func (i immutableScopeStorage) duplicate() immutableScopeStorage {
+	other := make(map[string][]byte, len(i))
+	for k, v := range i {
+		other[k] = v
+	}
+	return other
+}
+
+func RecorderPermissionValidatorFuzzer(t *testing.T, randBytes []byte) {
+	require := require.New(t)
+	// create a set of keys which would be used for testing.
+	// half of these keys would "exists", where the other won't.
+	existingKeys := createKeys(0)
+	nonExistingKeys := createKeys(1)
+	existingKeyValue := createKeysValues(existingKeys)
```

Comment:
Won't this always create the same set of existing / non-existing keys?

Would it make sense to add additional arguments to the fuzzing function to receive random inputs rather than setting them up ourselves?

For example, we can add a single byte argument to get a random input and then use `b` and `b+1` to create existing and non-existing keys.

We should be able to do the same with the operation count as well.